pallets/storage-proofs/src/lib.rs

```diff
@@ -462,8 +418,17 @@ pub trait FileSystemInterface {
         + MaxEncodedLen
         + FullCodec;
 
-    /// Check if a Merkle Patricia Forest root hash belongs to a given Storage Provider.
-    fn is_valid_root_for_sp(who: Self::StorageProvider, root: Self::ForestRoot) -> bool;
+    /// Check if an account is a registered Provider.
+    fn is_provider(who: Self::Provider) -> bool;
+
+    // Get Provider from AccountId, if it is a registered Provider.
+    fn get_provider(who: Self::AccountId) -> Option<Self::Provider>;
+
+    /// Get the root for a registered Provider.
+    fn get_root(who: Self::Provider) -> Option<Self::MerkleHash>;
+
+    /// Get the stake for a registered  Provider.
+    fn get_stake(who: Self::Provider) -> Option<Self::Balance>;
```

Comment:
This is the return type I mean. I think, for it to be a value and not a type or trait, it can't have all three required traits specified above. I might be mistaken though, let's chat about this.